include/ocpp/v16/smart_charging.hpp

```diff
@@ -39,11 +40,11 @@ class SmartChargingHandler {
 private:
     std::map<int32_t, std::shared_ptr<Connector>> connectors;
     std::shared_ptr<ocpp::v16::DatabaseHandler> database_handler;
+    std::shared_ptr<ChargePointConfiguration> configuration;
```

Comment:
There are multiple places where `shared_ptr` is used but not the best solution. It can actually be a reference `ChargePointConfiguration&` since the configuration object will outlive the smart_charging_handler . We plan to do a general refactor to reduce the amount of `shared_ptr` when they are not required.